dlpy/transformers/bert_model.py

```diff
@@ -0,0 +1,961 @@
+
+        Parameters
+        ----------
+        num_target_var: int
+            Specifies the number of target variables.
+        
+        Returns
+        -------
+        :class:`DataSpec`
+            
+        '''
+        
+        return create_data_spec(self.ds_layers, self._classification_problem, num_target_var)
+        
+    def set_optimizer_parameters(self, beta1=0.9, beta2=0.999, learning_rate=3e-5, max_epochs=3, mini_batch_size=1):
+        '''
+        Sets optimizer parameters according to defaults in original BERT paper.
+
+        Parameters
+        ----------
+        beta1 : double, optional
+            Specifies the exponential decay rate for the first moment in
+            the Adam learning algorithm.
+            Default: 0.9
+        beta2 : double, optional
+            Specifies the exponential decay rate for the second moment in
+            the Adam learning algorithm.
+            Default: 0.999
+        learning_rate : double, optional
+            Specifies the learning rate.
+            Default: 3e-5
+        max_epochs : int, optional
+            Specifies the maximum number of epochs.
+            Default: 3
+        mini_batch_size : int, optional
+            Specifies the number of observations per thread in a mini-batch.
+            Default: 1
+            
+        '''
+    
+        # the optimizer dictionary is set according to the parameter choices recommended
+        # for fine-tuning in the original BERT paper
+        if self._verbose:
+            log_level = 3
+        else:
+            log_level = 0
+            
+        # determine what to train
+        if self._freeze_base:
+            freeze_layers_to = self._rnn_layer
+            freeze_layers = None
+        else:
+            freeze_layers_to = None
+            freeze_layers = [self._rnn_layer]
+                
+        # set optimizer parameters similar to settings in original BERT paper
+        self._optimizer = Optimizer(algorithm=AdamSolver(beta1=beta1, beta2=beta2, learning_rate=learning_rate,
+                                                         gamma=0.0, power=0.0),
+                                    mini_batch_size=mini_batch_size, seed=self.seed, max_epochs=max_epochs,
+                                    dropout_type='inverted', log_level=log_level, freeze_layers_to=freeze_layers_to,
+                                    freeze_layers=freeze_layers)
+                                   
+    def get_tokenizer(self):
+        '''
+        Retrieves _tokenizer used by BERT model
+        
+        Returns
+        -------
+        :class:`BertTokenizer`
+            
+        '''
+        
+        return self._tokenizer
+
+    def get_max_sequence_length(self):
+        '''
+        Extracts the maximum sequence length supported by BERT model
+        
+        Returns
+        -------
+        int - xaximum sequence length supported by BERT model
```

Comment:
can you please fix the typo.

Reply:
I will fix that with my next pull request if that is OK with you.  I am currently making changes to support distilBERT and RoBERTa.  These are not trivial changes, and bert_model.py is impacted.  I'd rather not have to back out all my changes to fix something this minor.

Reply:
when you do a commit it should reflect this change. that said, if this is troublesome - I am happy to wait.

Reply:
I made the change.